flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/stream/table/JoinITCase.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.runtime.stream.table
+
+import org.apache.flink.api.scala._
+import org.apache.flink.streaming.api.TimeCharacteristic
+import org.apache.flink.table.api.scala._
+import org.apache.flink.streaming.api.scala.{DataStream, StreamExecutionEnvironment}
+import org.apache.flink.streaming.util.StreamingMultipleProgramsTestBase
+import org.apache.flink.table.api.{TableEnvironment, TableException}
+import org.apache.flink.table.runtime.utils.StreamITCase
+import org.apache.flink.types.Row
+import org.junit.Test
+
+/**
+  * Test if the table joins can be correctly translated.
+  */
+class JoinITCase extends StreamingMultipleProgramsTestBase {
```

Comment:
ITCases are quite expensive to run. That's why we try to avoid them if possible and if we add them, they should test as much as possible (i.e., result correctness) to justify the additional build time.

However, I don't think we need ITCases for this feature. The correct execution is already validated by harness tests and SQL ITCases. It is sufficient to add plan validation tests, i.e., tests that check the correct translation of Table API queries to execution plans (without executing them). Have a look at `org.apache.flink.table.api.stream.table.AggregateTest` as an example. Since SQL and Table API have a common execution layer, validating the plans is sufficient.

Reply:
Thanks for pointing this out, @fhueske. I'll rework the test.